### PR TITLE
SQL database: included_columns option

### PR DESCRIPTION
**Pull request by steinitzu**
### Tell us what you do here
- improving, documenting, or customizing an existing source (please link an issue or describe below)

### Short description

This adds `included_columns` option for sql database ( I would have named it `columns` but that was conflicting with the `$COLUMNS` shell variable in config resolution)   

Can be set with argument or config in in the standalone resource, and with config in the database source, e.g.:

```toml
[sources.sql_database.campaign]
included_columns = ["id", "name", "created_at", "updated_at"]
```

### Related Issues
One part of https://github.com/dlt-hub/verified-sources/issues/531#issuecomment-2267646842  
The query editing callback I would do in another PR

### CODE REVIEW

File: sources/sql_database/__init__.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def sql_database(
             name=table.name,
             primary_key=get_primary_key(table),
             spec=SqlDatabaseTableConfiguration,
-            columns=table_to_columns(table, reflection_level, type_adapter_callback),
+            # columns hint will be set at runtime, after included_columns setting has been resolved
```
Comment (steinitzu) [inline sources/sql_database/__init__.py:114]: @rudolfix is there any sense in trying to resolve the config there before resource is called? The column hints are now added with filtered columns in `table_rows` (same as with deferred reflection mode before)

Comment (rudolfix) [reply sources/sql_database/__init__.py:114]: hmmm `sql_table` actually resolves config before we create resource. maybe we just call it here and yield what it returns?

File: sources/sql_database/__init__.py [hunk 2]
```diff
@@ -142,6 +142,7 @@ def sql_table(
     table_adapter_callback: Callable[[Table], None] = None,
     backend_kwargs: Dict[str, Any] = None,
     type_adapter_callback: Optional[TTypeAdapter] = None,
+    included_columns: Optional[List[str]] = None,
```
Comment (rudolfix) [inline sources/sql_database/__init__.py:145]: maybe we should just name it columns? WDYT?

File: sources/sql_database/schema_types.py [hunk 3]
```diff
@@ -139,13 +139,15 @@ def table_to_columns(
...
     type_conversion_fallback: Optional[TTypeAdapter] = None,
+    included_columns: Optional[List[str]] = None,
 ) -> TTableSchemaColumns:
     """Convert an sqlalchemy table to a dlt table schema."""
     return {
         col["name"]: col
         for col in (
             sqla_col_to_column_schema(c, reflection_level, type_conversion_fallback)
             for c in table.columns
+            if included_columns is None or c.name in included_columns
```
Comment (rudolfix) [inline sources/sql_database/schema_types.py:150]: this certainly works. maybe we could remove columns from the Table object? so for example when table adapter is called user sees right set of columns?

or rename included_columns to selected_columns :)

**Review by rudolfix — CHANGES_REQUESTED** [review]
this is almost LGTM!

File: sources/sql_database/__init__.py (hunk 1 above)
Comment (steinitzu) [reply sources/sql_database/__init__.py:114]: Hmm maybe that would work. I'll try it.

File: sources/sql_database/__init__.py (hunk 2 above)
Comment (steinitzu) [reply sources/sql_database/__init__.py:145]: I wanted to but when there's a `COLUMNS` shell/env variable then everything fails. The config resolver tries to read that as a fallback when you don't set the option.

File: sources/sql_database/schema_types.py (hunk 3 above)
Comment (steinitzu) [reply sources/sql_database/schema_types.py]: yep deleting the sqla columns is simpler. Doing this now in the default table adapter https://github.com/dlt-hub/verified-sources/pull/550/files#diff-5d691408095c848d7a6e68ba31e82c21c396fb280c655ad5f09d68f579bba94eR41

File: sources/sql_database/__init__.py (hunk 1 above)
Comment (steinitzu) [reply sources/sql_database/__init__.py]: Ok re-using `sql_table` resource now, seems to work fine and reduces duplication a bit.

**Comment by steinitzu** [comment]
@rudolfix please check the updated version and let me know if any other changes needed

**Review by rudolfix — APPROVED** [review]
LGTM!

**Pull request merged**